sm/sm.go

```diff
@@ -66,3 +70,29 @@ func setUpLogging(logLevel string, logFormat string) {
 		logrus.SetFormatter(&logrus.TextFormatter{})
 	}
 }
+
+func getK8SConfigFile() *env.ConfigFile {
+	configFileLocation, err := k8senv.GetMountPath(k8senv.K8SConfigLocationEnvVarName)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+	return &env.ConfigFile{
+		Path:   configFileLocation,
+		Name:   "application",
+		Format: "yaml",
+	}
+}
+
+// GetEnvironment returns the specific server.Environment required to run the Service Manager
+func GetEnvironment() server.Environment {
+	envFlag := os.Getenv("SM_RUN_ENV")
+
+	switch envFlag {
```

Comment:
Different environments cannot exist together. So it will be `map[string]func() env` or something similar (for example an object having `Get` method). Then if no value is returned we have to create a default env. While this is possible I don't see any benefits in doing so. The code will be more and not clearer imo. Do you see any problems with the current approach? I agree though that having switch/case is not very good if we have many platforms.